File: algorithms/[781]森林中的兔子/solution.ts
// topics = ["哈希表"]

function numRabbits(answers: number[]): number {
  const counter = new Map<number, number>();
  for (const ans of answers) {
    counter.set(ans, (counter.get(ans) ?? 0) + 1);
  }

  let ans = 0;
  for (const [k, v] of counter.entries()) {
    ans += Math.ceil(v / (k + 1)) * (k + 1);
  }

  return ans;
}

File: algorithms/[781]森林中的兔子/solution.py
# topics = ["哈希表"]

from math import ceil
from collections import Counter
from typing import Dict, List


class Solution:
    def numRabbits(self, answers: List[int]) -> int:
        d: Dict[int, int] = Counter(answers)

        res = 0
        for k, v in d.items():
            res += ceil(v / (k + 1)) * (k + 1)

        return res
